### PR TITLE
[Noetic] Added Ignition common profiler to gazebo_plugins

### PR DESCRIPTION
**Pull request by ahcorde**
This PR is related to this other in Gazebo https://github.com/osrf/gazebo/pull/2776

It will allow to profile the methods in `gazebo_plugins` which are called periodically

To compile with the profile enable you need add this option `-DENABLE_PROFILER=1`. For example:

```
catkin_make_isolated --pkg gazebo_plugins --cmake-args -DENABLE_PROFILER=1
```

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_prosilica.cpp [hunk 1]
```diff
@@ -150,8 +154,12 @@ void GazeboRosProsilica::OnNewImageFrame(const unsigned char *_image,
       {
         if (sensor_update_time - this->last_update_time_ >= this->update_period_)
         {
+          IGN_PROFILE_BEGIN("PutCameraData");
           this->PutCameraData(_image, sensor_update_time);
+          IGN_PROFILE_END();
+          IGN_PROFILE_BEGIN("PutCameraData");
```
Comment (iche033) [inline gazebo_plugins/src/gazebo_ros_prosilica.cpp]: `PublishCameraInfo`

**Review by iche033 — APPROVED** [review]
works for me. Just one minor typo.

**Comment by ahcorde** [comment]
This test is failing `_ctest_gazebo_plugins_rostest_test_range_range_plugin.test (Failed)`. It looks unrelated. Can we merge this @iche033 ?

**Comment by iche033** [comment]
looks like that test also failed before. Could be flaky

**Review by iche033 — APPROVED** [review]

**Comment by eborghi10** [comment]
@ahcorde is there a reason why this was not added to ROS Melodic?

**Comment by ahcorde** [comment]
@eborghi10 not really, I will be happy to review a the PR if you want to contribute

**Comment by eborghi10** [comment]
> @eborghi10 not really, I will be happy to review a the PR if you want to contribute

@ahcorde, thanks for answering quickly. I tried to make it work yesterday but I didn't see the information from the new plugins in the Profiler so I wondered if it's related to how `gazebo/common` works in comparison to Ignition Common.

At least for now, I opened a PR in Gazebo to use the profiler since it wasn't being installed from source: <https://github.com/osrf/gazebo/pull/3032>.

I'll try next week to create a PR with the changes. It shouldn't be hard.

**Pull request merged**